src/core/radio/radio.hpp

```diff
@@ -408,11 +408,22 @@ class Radio : public InstanceLocator, private NonCopyable
     /**
      * This method updates the CSL sample time in radio.
      *
-     * @param[in]  aCslSampleTime  The CSL sample time.
+     * @param[in]  aCslSampleTime   The CSL sample time.
      *
      */
     void UpdateCslSampleTime(uint32_t aCslSampleTime);
 
+#if OPENTHREAD_CONFIG_MAC_SOFTWARE_RX_TIMING_ENABLE
+    /**
+     * This method updates the CSL start time and duration in radio.
+     *
+     * @param[in]  aCslSampleStart  The CSL sample start time.
+     * @param[in]  aCslSampleDur    The CSL sample duration.
+     *
+     */
+    void UpdateCslSampleStart(uint32_t aCslSampleStart, uint32_t aCslSampleDur);
```

Comment:
Looks more like updating a sample window(start and end) than just start to me
```suggestion
    void UpdateCslSampleWindow(uint32_t aCslSampleStart, uint32_t aCslSampleDur);
```